File: slurm_code/a.add_shell_info.py
#!usr/bin/python3
import numpy as np
import pandas as pd
import sys

fi = pd.read_table(sys.argv[1], header = None)

for i in range(len(fi)):
    print ('echo Job' + str(i))
    print ('echo "Job'+str(i)+' start: `date`"')
    print ('if')
    print ('\t'+ str(fi[0].iloc[i]))
    print ('then')
    print ('\techo "Job'+str(i)+' done:`date`"')
    print ('else')
    print ('\techo "ERROR in: "' + str(fi[0].iloc[i]))
    print ('\texit')
    print ('fi')

File: slurm_code/a.add_shell_info.pl
#!usr/bin/perl
use strict;
use warnings;

my $num = 0;
open IN, "<$ARGV[0]" or die $!;
while(<IN>){
	chomp;
	my $line = $_;
	$num ++;	
	print "echo Job$num\n";
	print "echo Job$num start:`date`\n";
	print "if\n";
	print "\t$line\n";
	print "then\n";	
	print "\techo Job$num done:`date`\n";
	print "else\n";
	print "\techo ERROR in: $line\n";	
	print "\texit\n";
	print "fi\n";
}
